src/models.py

```diff
@@ -18,3 +26,39 @@ def __str__(self) -> str:
         return "-to-".join(
             [self.start.strftime("%Y-%m-%d"), self.end.strftime("%Y-%m-%d")]
         )
+
+
+class Token:
```

Comment:
We might also call it ERC20Token, if we always want to have an address. If we also wanna allow it to represent native eth token - what I think you are doing - then we could also make the address an enum that is either NATIVE or Address(address).

If we then later have choose the token in the Transfer class, then we also don't need a transfer_type, as we can just judge it based on the enum: If the token is NATIVE, you do a native transfer, otherwise an ERC20 transfer.

Reply:
We don't actually use this class at all for native transfers the token field is optional and None represents a native transfer. 

I can, however call this ERC20 token, but at this point there isn't any other kind of token, so I am not sure distinguishing between different tokens is necessary until there is at least one other kind of token.

The following PR (right after this) removes the `token_type` since the Transfer type (ERC20 or Native) is uniquely determined by the `token: Optional[Token]` field.